snapshot/android/utils.js

```diff
@@ -1,17 +1,66 @@
 const { chmodSync, createWriteStream, existsSync } = require("fs");
 const { tmpdir, EOL } = require("os");
-const { dirname, join } = require("path");
+const { join, relative, isAbsolute } = require("path");
+const os = require("os");
 
 const { mkdir } = require("shelljs");
 const { get } = require("request");
 const { getProxySettings } = require("proxy-lib");
+const semver = require("semver");
 
 const CONSTANTS = {
     SNAPSHOT_TMP_DIR: join(tmpdir(), "snapshot-tools"),
+    MAC_OS_NAME: "darwin",
+    WIN_OS_NAME: "win",
+    LINUX_OS_NAME: "linux"
 };
 
 const createDirectory = dir => mkdir('-p', dir);
 
+function getHostOS() {
+    const hostOS = os.type().toLowerCase();
+    if (hostOS.startsWith(CONSTANTS.MAC_OS_NAME))
+        return CONSTANTS.MAC_OS_NAME;
+    if (hostOS.startsWith(CONSTANTS.LINUX_OS_NAME))
+        return CONSTANTS.LINUX_OS_NAME;
+    if (hostOS.startsWith(CONSTANTS.WIN_OS_NAME))
+        return CONSTANTS.WIN_OS_NAME;
+    return hostOS;
+}
+
+function getHostOSVersion() {
+    return os.release();
+}
+
+function getHostOSArch() {
+    return os.arch();
+}
+
+function has32BitArch(targetArchs) {
+    return Array.isArray(targetArchs) && targetArchs.some(arch => arch === "arm" || arch === "ia32")
```

Comment:
Do we support `targetArchs` to be a string? If we support it, we need to extend the check: 
```suggestion
    return (Array.isArray(targetArchs) && targetArchs.some(arch => arch === "arm" || arch === "ia32")) ||
    (targetArchs === "arm" || targetArchs === "ia32");
```